### ISSUE TITLE
Elementor texturize the content output by shortcodes

### ISSUE THREAD

**Issue by Tofandel** [elementor/elementor#9340]
Some shortcodes have some fancy content (like angular templates in them) and texturizing them break them completely

The bug originates from wp-content/plugins/elementor/includes/base/widget-base.php line 373

`$content = do_shortcode( $content );
$content = wptexturize( $content );
`

The order there should be inverted
`$content = do_shortcode( wptexturize(  $content ) );`

**Comment by nextend** [comment]
This happens with Smart Slider 3 too. Some of the && characters in the script tag turns into &#038;&#038;

**Comment by Tofandel** [comment]
@shilo-ey Any chance you could review this?.. Not being able to insert some shortcode in tabs and other elementor widgets is an annoying bug that could be easily fixed

**Comment by tiendungdev** [comment]
Waiting for patches

**Comment by Tofandel** [comment]
Don't hold your breath.. An advice, steer away from elementor

**Comment by tiendungdev** [comment]
> Don't hold your breath.. An advice, steer away from elementor

Which plugin builder should I choose?

**Comment by Tofandel** [comment]
I would say Beaver

**Comment by vkurko** [comment]
This Elementor's bug affects Bookly plugin too.

The `wptexturize` is designed to be called before the `do_shortcode`. You can clearly see this in the `wptexturize` [documentation](https://developer.wordpress.org/reference/functions/wptexturize/). The function uses `no_texturize_shortcodes`  filter to skip certain shortcodes, which means that it expects shortcodes to be passed in with the input content. The Elementor breaks this behaviour. 

You can also refer to this [ticket](https://core.trac.wordpress.org/ticket/6969) which resulted in the `no_texturize_shortcodes` filter being created. As you can see the idea was that if plugins want their contents formatted, they can pass the contents to the `wptexturize` function itself. Unfortunately the Elementor does not follow this rule and breaks third-party plugins.

@KingYes please consider a solution to this problem.

**Comment by Tofandel** [comment]
Yes bookly is exactly the plugin I had problems with because it uses angular templates in shortcodes which are not output correctly once wp_texturized.

KingYes didn't even look at this, he just closed the PR with no comment. Again if you can, change page builder because elementor has no future.

**Comment by CKMacLeod** [comment]
The ability to remove wptexturize() for whatever reason is built into core WordPress. It should be made possible/easy in Elementor, too. Right now, the easiest, direct way to handle the issue is to go to widget-base.php and comment out line 373, under the function parse_text_editor(). There's a content filter in this function, but it's located at the top, before several other $content functions are applied. It wouldn't be hard - or break anything - to add a filter either before returning $content or to return $content with a filter, like `return apply_filters( 'rendered_content', $content, $this->get_settings() ) ;`

Alternatively, the option to turn off wptexturize could be added to the settings, but doing so would obviously be more complicated.

**Comment by gmariani** [comment]
Breaks envira gallery  albums too
